Source/Core/DolphinQt/HotkeyScheduler.cpp

```diff
@@ -525,45 +530,51 @@ void HotkeyScheduler::Run()
       Config::SetCurrent(Config::GFX_STEREO_CONVERGENCE,
                          std::min(stereo_convergence + 5, Config::GFX_STEREO_CONVERGENCE_MAXIMUM));
 
-    // Freelook
-    static float fl_speed = 1.0;
-
     if (IsHotkey(HK_FREELOOK_TOGGLE))
     {
       const bool new_value = !Config::Get(Config::GFX_FREE_LOOK);
       Config::SetCurrent(Config::GFX_FREE_LOOK, new_value);
       OSD::AddMessage(StringFromFormat("Freelook: %s", new_value ? "Enabled" : "Disabled"));
     }
 
-    if (IsHotkey(HK_FREELOOK_DECREASE_SPEED, true))
-      fl_speed /= 1.1f;
+    if (g_Config.bFreeLook)
+    {
+      // Freelook
+      static float fl_speed = 1.0;
+
+      if (IsHotkey(HK_FREELOOK_DECREASE_SPEED, true))
+        fl_speed /= 1.1f;
 
-    if (IsHotkey(HK_FREELOOK_INCREASE_SPEED, true))
-      fl_speed *= 1.1f;
+      if (IsHotkey(HK_FREELOOK_INCREASE_SPEED, true))
+        fl_speed *= 1.1f;
 
-    if (IsHotkey(HK_FREELOOK_RESET_SPEED, true))
-      fl_speed = 1.0;
+      if (IsHotkey(HK_FREELOOK_RESET_SPEED, true))
+        fl_speed = 1.0;
 
-    if (IsHotkey(HK_FREELOOK_UP, true))
-      VertexShaderManager::TranslateView(0.0, 0.0, -fl_speed);
+      if (IsHotkey(HK_FREELOOK_UP, true))
+        VertexShaderManager::TranslateView(0.0, 0.0, -fl_speed);
 
-    if (IsHotkey(HK_FREELOOK_DOWN, true))
-      VertexShaderManager::TranslateView(0.0, 0.0, fl_speed);
+      if (IsHotkey(HK_FREELOOK_DOWN, true))
+        VertexShaderManager::TranslateView(0.0, 0.0, fl_speed);
 
-    if (IsHotkey(HK_FREELOOK_LEFT, true))
-      VertexShaderManager::TranslateView(fl_speed, 0.0);
+      if (IsHotkey(HK_FREELOOK_LEFT, true))
+        VertexShaderManager::TranslateView(fl_speed, 0.0);
 
-    if (IsHotkey(HK_FREELOOK_RIGHT, true))
-      VertexShaderManager::TranslateView(-fl_speed, 0.0);
+      if (IsHotkey(HK_FREELOOK_RIGHT, true))
+        VertexShaderManager::TranslateView(-fl_speed, 0.0);
 
-    if (IsHotkey(HK_FREELOOK_ZOOM_IN, true))
-      VertexShaderManager::TranslateView(0.0, fl_speed);
+      if (IsHotkey(HK_FREELOOK_ZOOM_IN, true))
+        VertexShaderManager::TranslateView(0.0, fl_speed);
 
-    if (IsHotkey(HK_FREELOOK_ZOOM_OUT, true))
-      VertexShaderManager::TranslateView(0.0, -fl_speed);
+      if (IsHotkey(HK_FREELOOK_ZOOM_OUT, true))
+        VertexShaderManager::TranslateView(0.0, -fl_speed);
 
-    if (IsHotkey(HK_FREELOOK_RESET, true))
-      VertexShaderManager::ResetView();
+      if (IsHotkey(HK_FREELOOK_RESET, true))
+        VertexShaderManager::ResetView();
+
+      const auto gyro_amt = GetGyro(HGGP_FREELOOK_ROTATE);
+      VertexShaderManager::RotateView(gyro_amt.x, gyro_amt.y, gyro_amt.z);
```

Comment:
Yaw and roll are apparently swapped in `IMUGyroscope`, so Y and Z need to be swapped here.

Reply:
Oh gosh, that explains so much!  I wondered why Dolphin wasn't behaving as expected...